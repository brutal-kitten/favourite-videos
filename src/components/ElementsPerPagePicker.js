import React, { Component } from 'react';

class ElementsPerPagePicker extends Component {

  state = {
    value: 10
  }

  handleChange = (event) => {

    event.preventDefault();
    console.log(event.target.value);
    this.setState({value: event.target.value});
    this.props.setIndexes(event.target.value);
  }



  render() {

    return (
      <div className="elementsPerPagePicker">
        <span className="pickerText">Elements on page</span>
        <select id="selectNumber" value={this.state.value} onChange={(event) => this.handleChange(event)}>
          <option value="10">10</option>
          <option value="8">8</option>
          <option value="6">6</option>
          <option value="4">4</option>
          <option value="2">2</option>
        </select>
      </div>
    )
  }
}

export default ElementsPerPagePicker;
