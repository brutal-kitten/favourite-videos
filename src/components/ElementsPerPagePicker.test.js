import React from 'react';
import ReactDOM from 'react-dom';
import ElementsPerPagePicker from './ElementsPerPagePicker.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <ElementsPerPagePicker />).toJSON();
  expect(tree).toMatchSnapshot();
});

const mockSetIndexes = jest.fn();
const props = {
  setIndexes: mockSetIndexes,
  elementsPerPage: 10
};

const wrapper = mount(<ElementsPerPagePicker {...props}/>);

it('has proper value given by props', () => {
  expect(wrapper.props().elementsPerPage).toBe(10);
})


describe('selection works properly', () => {

  wrapper.find('select').simulate('change', {
    target: {
      value: 6
    },
    preventDefault: () => {},
  });

  it('should call mockSort function with 1 given value', () => {
    expect(mockSetIndexes.mock.calls.length).toBe(1);
    expect(mockSetIndexes.mock.calls[0][0]).toBe(6);
  });

  it('should have 5 option', () => {
    expect(wrapper.find('select').children().length).toBe(5);
  });

})
