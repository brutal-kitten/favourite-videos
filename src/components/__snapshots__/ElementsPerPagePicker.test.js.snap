// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="elementsPerPagePicker"
>
  <span
    className="pickerText"
  >
    Elements on page
  </span>
  <select
    id="selectNumber"
    onChange={[Function]}
    tabIndex="0"
  >
    <option
      value="10"
    >
      10
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="2"
    >
      2
    </option>
  </select>
</div>
`;
